addons/sale_purchase/models/sale_order_line.py

```diff
@@ -226,14 +227,19 @@ def _purchase_service_match_supplier(self, warning=True):
             raise UserError(_("There is no vendor associated to the product %s. Please define a vendor for this product.", self.product_id.display_name))
         return suppliers[0]
 
+    def _get_additional_domain_for_purchase_order_line(self):
+        return  [('sale_order_id', '=', self.order_id.id)]
+
     def _purchase_service_match_purchase_order(self, partner, company=False):
         return self.env['purchase.order.line'].search(
-            [
-                ('partner_id', '=', partner.id),
-                ('state', '=', 'draft'),
-                ('company_id', '=', (company and company or self.env.company).id),
-                ('sale_order_id', '=', self.order_id.id),
-            ],
+            Domain.AND([
```

Comment:
`AND` does not exist
```suggestion
            Domain.and([
```

Reply:
Actually, the method has been renamed. 😅 

Difficult to follow the framework changes. 🙈